pkg/daemon/config.go

```diff
@@ -184,6 +185,93 @@ func getSource(isTs2phcMaster string) event.EventSource {
 	return event.PPS
 }
 
+type synceToIfaceRelation struct {
+	synceDevice string
+	ifaceNames  []string
+	clockId     string
+}
+
+type synceRelations struct {
+	devices []*synceToIfaceRelation
+}
+
+// extractSynceRelations extracts relation of synce device to interfaces
+// The sections are ordered in the following way:
+//  1. Device section specifies the configuration of a one logical device e.g. 'synce1'.
+//     The name must be enclosed in extra angle bracket when defining new device section e.g. [<synce1>]
+//     All ports defined by port sections after the device section will create one SyncE device
+//     (until next device section).
+//  2. Port section - any other section not starting with < (e.g. [eth0]) is the port section.
+//     Multiple port sections are allowed. Each port participates in SyncE communication.
+//  3. Any other section not starting with { (e.g. {SMA1}) is the external source section.
+//     Multiple external source sections are allowed. Each external source participates in best
+//     source selection algorithm for EEC
+func (conf *ptp4lConf) extractSynceRelations() *synceRelations {
+	var deviceName string
+	r := &synceRelations{
+		devices: []*synceToIfaceRelation{},
+	}
+	ifaces := []string{}
+	re, _ := regexp.Compile(`[{}<>\[\] ]+`)
+
+	for _, section := range conf.sections {
+		if strings.HasPrefix(section.sectionName, "[<") {
+			deviceName = re.ReplaceAllString(section.sectionName, "")
+			ifaces = []string{}
+		} else if strings.HasPrefix(section.sectionName, "[{") {
+			binding := synceToIfaceRelation{
+				synceDevice: deviceName,
+				ifaceNames:  ifaces,
```

Comment:
How do we know that ifaces will already be set?

Reply:
According to the comment above (regarding the config file section order), interface sections follow the device section (prefix "[<") and are followed by the port section (prefixed by  "[{" ). At the moment we reach this line, all the interfaces related to this device should be already processed (line 229).

Reply:
Thanks, I found a bug and fixed it. First of all, the external source section is optional - it might not be there for non-GM settings. But it is a requirement in synce4l documentation that device is followed by the interfaces.
Added a couple of unit tests to verify nothing breaks if the config file is wrong, and a log that says that clock id is not found because the interfaces are missing